docs/madr/decisions/049-meshexternalservice-api.md

```diff
@@ -0,0 +1,492 @@
+        subjectAltNames: # if subjectAltNames is not defined then take domain or ips
+          - example.com
+          - "spiffe://example.local/ns/local"
+        caCert: 
+          inline: 123
+        clientCert:
+          secret: 123
+        clientKey:
+          secret: 123
+status: 
+  vip:
+    value: 242.0.0.1
+    type: Kuma
+  addresses:
+  - hostname: example.ext.svc.local
+    status: NotAvailable
+    origin:
+      kind: HostnameGenerator
+      name: k8s-example-service-hostname
+    reason: "addresses are overlapping with ext2"
+```
+* **spec**:
+  * **match**: defines a traffic that should be routed through the sidecar
+    * **type**: type of the match, one of `KumaHostname`, `Domain`, `CIDR` and `IP` are available
+      * `KumaHostname`: allocates a VIP for a domain provided in the `value` field
+      * `Domain`: handles traffic to the specified domain
+      * `CIDR`: handles traffic to specified addresses range
+      * `IP` handles the traffic to specified IP
+    * **value**: depends of the type can be an existing domain, new domain name, CIDR or IP
+    * **port**: defines a port to which a user does requests
+    * **protocol**: defines a protocol of the communication. Possible values:
+      * `tls`: should be used when TLS traffic is originated by the client application in the case the `kuma.io/protocol` would be tcp
+      * `tcp`: WARNING: shouldn't be used when match has only domains. On the TCP level we are not able to disinguish domain, in this case it is going to hijack whole traffic on this port. We are going to validate configuration and do not apply config when protocol is tcp and `type: Domain`.
+      * `grpc`
+      * `http`
+      * `http2`
+  * **type**: defines what kind of destination it is, one of `Managed`, `Passthrough`, or `Extension`, (Default: `Passthrough`)
+    * `Managed`: allows creating a set of destination endpoints and `TLS` configuration, when defined section `extension` is not available.
+  * **managed**: defines where matched requests should be routed, only for a `type: Managed`
+    * **extension**: struct for a plugin configuration, only for a `type: Extension`
+      * **type**: defines what kind of plugin to use, it's a string type so any new plugins should works.
+      * **config**: json map that is mapped to configuration provided in the type.
+    * **endpoints**: defines a list of endpoints
+      * **address**: defines an address to which a user want to send a request. Is possible to provide `domain`, `ip` and `unix` sockets
+      * **port**: defines a port of a destination.
+    * **tls**: provides a TLS configuration when proxy is resposible for a TLS origination
+      * **version**: section for providing version specification.
+        * **min**: defines minmum supported version. One of `TLS_AUTO`, `TLSv1_0`, `TLSv1_1`, `TLSv1_2`, `TLSv1_3`
+        * **max**: defines maximum supported version. One of `TLS_AUTO`, `TLSv1_0`, `TLSv1_1`, `TLSv1_2`, `TLSv1_3`
```

Comment:
The default for both is `TLS_AUTO`, right?

Reply:
That's not the way we do enum usually as we follow k8s conventions

Reply:
I'll rename them